packages/auth/src/components/EmailPasswordSignUpForm/index.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { showAppMessage, showAppSpinner, hideAppSpinner } from '../../../../app/src/actions';
+import useFirebaseApp from '../../../../firebase-hooks/src/useFirebaseApp';
+import { nameIcon, emailIcon, passwordIcon } from './icons';
+import { StyledInput, StyledButton } from './styled';
+
+const nameLabel = <FormattedMessage id="EmailPasswordSignUpForm.name" defaultMessage="Name" />;
+const emailLabel = (
+    <FormattedMessage id="EmailPasswordSignUpForm.email" defaultMessage="Email Address" />
+);
+const passwordLabel = (
+    <FormattedMessage id="EmailPasswordSignUpForm.password" defaultMessage="Password" />
+);
+const buttonLabel = (
+    <FormattedMessage id="EmailPasswordSignUpForm.signup" defaultMessage="Create account" />
+);
+
+const EmailPasswordSignUpForm = (props) => {
+    const intl = useIntl();
+    const { className, style } = props;
+    const [name, setName] = useState('');
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const app = useFirebaseApp();
+
+    const namePlaceholder = intl.formatMessage({
+        id: 'EmailPasswordSignUpForm.namePlaceholder',
+        defaultMessage: 'Enter your name',
+    });
+    const emailPlaceholder = intl.formatMessage({
```

Comment:
can we move all these message components into a different file e.g. message.js then you can export then and that way the logic for I18n is isolated?

Reply:
The <FormatedMessage /> components above could be moved out of this file also

Reply:
> can we move all these message components into a different file e.g. message.js then you can export then and that way the logic for I18n is isolated?

we should not do that because `intl` is obtained with useIntl, and hooks can only be called within components, and the placeholder should be a plain string

Reply:
yeah, I hear you! ... I guess I was thinking about using `defineMessage` API. Why we need this is the question :) ... because moving forward I believe It will be nice to develop messages in a different file as a best practice, kinda separation of concern.

Reply:
by the way of example:

```js
import {defineMessages, defineMessage} from 'react-intl';

export const messages = defineMessages({
  greeting: {
    id: 'app.home.greeting',
    description: 'Message to greet the user.',
    defaultMessage: 'Hello, {name}!',
  },
});
export const msg = defineMessage({
  id: 'single'
  defaultMessage: 'single message',
  description: 'header'
})
```

then you can use the hook and do the formatting in the component

more info here: https://formatjs.io/docs/react-intl/api

Reply:
thanks for the clarification and the example, i think it is OK now